src/iosrv/vos.c

```diff
@@ -785,16 +786,20 @@ enum_unpack_recxs(daos_key_desc_t *kds, void *data,
 		return -DER_INVAL;
 
 	iod = &io->ui_iods[io->ui_iods_size - 1];
-	iod_akey = &iod->iod_name;
+	rc = daos_iov_copy(&iod_akey, &iod->iod_name);
+	if (rc)
+		return rc;
+
 	dkey = &io->ui_dkey;
-	if (dkey->iov_len == 0 || iod_akey->iov_len == 0) {
+	if (dkey->iov_len == 0 || iod_akey.iov_len == 0) {
 		rc = -DER_INVAL;
 		D_ERROR("invalid list buf %c\n", rc);
-		return rc;
+		D_GOTO(free, rc);
 	}
 
 	while (ptr < data + kds->kd_key_len) {
-		int		j = io->ui_iods_size - 1;
+		int		j = io->ui_iods_size == 0 ?
```

Comment:
Some confused, is it possible that io::ui_iods_size is zero? We have checked that at the line 785.

Reply:
there is complete_io in the while loop, which might make the iod_size to be 0, which is the key to fix the bug.